Automattic/newspack-ios
# Issues/91 new cells

Refs #91 

This PR is a first pass at cell UI for stories, text notes and photos, and stubbed cells for video and audio notes.

There is a lot of boiler plate in the changes (and project file and xib changes). Some highlights are:
- A CGSize extension for Hashable conformance.
- An ImageResizer utility used to downsize UIImages for display in cells.
- Thumbnail handling in the AssetsViewController
- The cells and cellprovider pattern (for decoupling the StoryAsset model from cells).
- StoryFolder helper methods and StoryAsset provider conformance.

Note: The behavior of the "cloud-upload" icons/buttons is not fully implemented. It will be revisited when syncing is addressed.  Also, video and audio cells are stubs and will see some modifications as video and audio features are implemented, so we'll skip the visual review for now.  Everything will get a final future design review closer to the v1 wrap up.

To test:
- Prep: Go to the Photos app. Select some photos and copy them.  Then use the files app to paste the photos in your current story folder.  This is to work around the known issue of imported images being all black, and will let you see the thumbnail images clearly. 
- Add some text notes and photos.
- Confirm the stories cell reflects the number of assets by asset type.
- Ensure cell layout looks good. (Nothing obviously broken.)
- Confirm cell colors look good in both light and dark mode.
- On the iPad, confirm that cells respect readable width margins.

@jleandroperez could I trouble you for a review of this one?  I apologize for the size, but about half of the changes are the xib and project file. 😬 

Examples:

<img width="386" alt="Screen Shot 2020-08-18 at 11 29 07 AM" src="https://user-images.githubusercontent.com/1435271/90540918-7cbe5f00-e147-11ea-8bbd-fa3136967d23.png">
<img width="407" alt="Screen Shot 2020-08-18 at 11 29 23 AM" src="https://user-images.githubusercontent.com/1435271/90540923-7f20b900-e147-11ea-9cf5-37a65db969a0.png">
<img width="392" alt="Screen Shot 2020-08-18 at 11 30 06 AM" src="https://user-images.githubusercontent.com/1435271/90540938-83e56d00-e147-11ea-8904-902ceed20533.png">
<img width="404" alt="Screen Shot 2020-08-18 at 11 29 40 AM" src="https://user-images.githubusercontent.com/1435271/90540944-8647c700-e147-11ea-9ddd-c1646341bae2.png">
<img width="386" alt="Screen Shot 2020-08-18 at 11 30 13 AM" src="https://user-images.githubusercontent.com/1435271/90540964-8ea00200-e147-11ea-950f-29cad37e25d6.png">
<img width="380" alt="Screen Shot 2020-08-18 at 11 29 54 AM" src="https://user-images.githubusercontent.com/1435271/90540976-92338900-e147-11ea-9975-c3524678b7ca.png">
